wfe2/wfe.go

```diff
@@ -2401,14 +2401,17 @@ func (wfe *WebFrontEndImpl) NewOrder(
 		return
 	}
 
-	refundLimits, err := wfe.checkNewOrderLimits(ctx, acct.ID, names, isRenewal)
-	if err != nil && features.Get().UseKvLimitsForNewOrder {
-		if errors.Is(err, berrors.RateLimit) {
-			wfe.sendError(response, logEvent, probs.RateLimited(err.Error()), err)
-			return
-		} else {
-			wfe.sendError(response, logEvent, web.ProblemDetailsForError(err, "While checking rate limits"), err)
-			return
+	var refundLimits func()
```

Comment:
Leaving this `nil` will result in a panic if it's called.

I was surprised that panic isn't surfaced by the tests. Looking below, we wouldn't call `refundLimits` unless the `ra.NewOrder` RPC fails. Perhaps we simply don't have a unittest that covers that case?

Recomendation: For now, change this to:

```
refundLimits := func() {}
```

In other words, set it to a non-`nil`, no-op function. That way it is guaranteed safe to call no matter which code path we take.

Also, please do a quick check of whether there is a unittest we would expect to cover the refunding case. If there is none, file a followup issue to add one. If there is one but it didn't cover the panic for some reason, we should figure out why before landing this change.

Reply:
Thanks for spotting this! I created #7903.